tests/test_stl.py

```diff
@@ -728,3 +729,13 @@ def test65_class_with_movable_field(clean):
         move_constructed=2,
         destructed=4
     )
+
+def test66_replace_extension():
```

Comment:
If you grep through for `Path` in the pybind11 test suite, there seems to be more material that isn't included in this PR (`test_fs_path()` in `test_stl.py`)